amethyst_network/src/simulation/message.rs

```diff
@@ -0,0 +1,21 @@
+use super::guarantees::{DeliveryGuarantee, OrderingGuarantee};
+use bytes::Bytes;
+
+/// Structure used to hold message payloads before they are consumed and sent by an underlying
+/// NetworkSystem.
+pub struct Message {
+    pub(crate) payload: Bytes,
+    pub(crate) delivery: DeliveryGuarantee,
+    pub(crate) ordering: OrderingGuarantee,
+}
+
+impl Message {
+    /// Creates and returns a new Message.
+    pub fn new(payload: &[u8], delivery: DeliveryGuarantee, ordering: OrderingGuarantee) -> Self {
```

Comment:
The expectation I have is that the end user will never be making this structure directly. This will only be made through the `NetworkSimulationResource`.